hawkular-alerts-rest-tests/src/test/groovy/org/hawkular/alerts/rest/CrossTenantITest.groovy

```diff
@@ -0,0 +1,109 @@
+        for (String tenantId : tenantIds) {
+            client.headers.put("Hawkular-Tenant", tenantId)
+            Trigger testTrigger = new Trigger("test-multiple-tenants", "CrossTenantITest");
+            def resp = client.delete(path: "triggers/test-multiple-tenants");
+            assert(200 == resp.status || 404 == resp.status)
+            resp = client.post(path: "triggers", body: testTrigger)
+            assertEquals(200, resp.status)
+
+            // ADD Firing condition
+            AvailabilityCondition firingCond = new AvailabilityCondition("test-multiple-tenants",
+                    Mode.FIRING, "test-multiple-tenants", AvailabilityCondition.Operator.NOT_UP);
+
+            Collection<Condition> conditions = new ArrayList<>(1);
+            conditions.add( firingCond );
+            resp = client.put(path: "triggers/test-multiple-tenants/conditions/firing", body: conditions)
+            assertEquals(200, resp.status)
+            assertEquals(1, resp.data.size())
+
+            // ENABLE Trigger
+            testTrigger.setEnabled(true);
+
+            resp = client.put(path: "triggers/test-multiple-tenants", body: testTrigger)
+            assertEquals(200, resp.status)
+
+            // SEND Data to generate Alerts
+            for (int i=0; i<numAlerts; i++) {
+                Data avail = new Data("test-multiple-tenants", System.currentTimeMillis() + (i*1000), "DOWN");
+                Collection<Data> datums = new ArrayList<>();
+                datums.add(avail);
+                resp = client.post(path: "data", body: datums);
+                assertEquals(200, resp.status)
+            }
+        }
+    }
+
+    @Test
+    void fetchEventsAndAlertsFromMultipleTenants() {
+        def start = System.currentTimeMillis();
+
+        List<String> tenantIds = ["tenant1", "tenant2", "tenant3", "tenant4"];
+        generateAlertsForMultipleTenants(tenantIds, 5);
+
+        client.headers.put("Hawkular-Tenant", "tenant1,tenant2,tenant3,tenant4")
+        def resp
+        for ( int i=0; i < 10; ++i ) {
+            Thread.sleep(500);
+
+            // FETCH recent alerts for trigger, there should be 5
+            resp = client.get(path: "admin/events", query: [startTime:start,triggerIds:"test-multiple-tenants"] )
+            if ( resp.status == 200 && resp.data.size() == 20 ) {
+                break;
+            }
+        }
+
+        // The alert processing happens async, so give it a little time before failing...
+        for ( int i=0; i < 10; ++i ) {
+            Thread.sleep(500);
+
+            // FETCH recent alerts for trigger, there should be 5
+            resp = client.get(path: "admin/alerts", query: [startTime:start,triggerIds:"test-multiple-tenants"] )
+            if ( resp.status == 200 && resp.data.size() == 20 ) {
+                break;
+            }
+        }
+        assertEquals(200, resp.status)
+        assertEquals(20, resp.data.size())
```

Comment:
It is tested on PersistenceTest#test0300GetAlertsMultipleTenants.
I haven't brought it to the REST ITest as sorting test are focus on PersistenceTest class.